### PR TITLE
Melee Combat Visuals

### PR DESCRIPTION
**Pull request by 77seven77seven**
## About the Pull Request

juicier blood effects and a blood effect like when being shot when hitting with a sharp weapon

https://github.com/Foundation-19/Foundation-19/assets/111151681/c52a8d05-c478-4f50-ab7e-1b0ebac7863f

## Why It's Good For The Game

It looks cooler and more responsive.

## Changelog

:cl:
add: Juicier sounds when stabbing someone.
add: Hitting someone with a sharp weapon now causes a blood overlay-like effect.
/:cl:

### CODE REVIEW

File: code/modules/mob/living/carbon/human/human_defense.dm [hunk 1]
```diff
@@ -254,6 +254,16 @@ meteor_act
 			if(BP_CHEST)
 				bloody_body(src)
 
+	//cool splat effect copied from projectile code
+	var/hit_dir = get_dir(attacker.loc, src)
+	var/obj/effect/decal/cleanable/blood/B = blood_splatter(get_step(src, hit_dir), src, 1, hit_dir)
+	B.icon_state = pick("dir_splatter_1","dir_splatter_2")
+	var/scale = min(1, round(effective_force / 50, 0.2))
```
Comment (aidan-bowman) [inline code/modules/mob/living/carbon/human/human_defense.dm]: You only use this var once, why not just move the calculation directly into `.Scale()`?

File: code/__defines/sound.dm [hunk 2]
```diff
@@ -58,6 +58,7 @@
 #define SFX_CHOP "chop"
 #define SFX_GLASS_CRACK "glasscrack"
 #define SFX_TRAY_HIT "tray_hit"
+#define SFX_WOUND_SPLATTER "splatter"
```
Comment (aidan-bowman) [inline code/__defines/sound.dm]:
```suggestion
#define SFX_STAB_SPLATTER "stab_splatter"
```

Don't forget to search and replace if you use this

File: code/modules/mob/living/carbon/carbon.dm [hunk 3]
```diff
@@ -320,6 +320,7 @@
 
 	//actually throw it!
 	visible_message(SPAN_WARNING("[message]"), range = min(itemsize*2,world.view))
+	src.do_attack_animation(target)
```
Comment (aidan-bowman) [inline code/modules/mob/living/carbon/carbon.dm]:
```suggestion
	do_attack_animation(target)
```

File: code/game/objects/items/weapons/material/material_weapons.dm [hunk 4]
```diff
@@ -2,7 +2,7 @@
 // This class of weapons takes force and appearance data from a material datum.
 // They are also fragile based on material data and many can break/smash apart.
 /obj/item/material
-	hitsound = 'sounds/weapons/bladeslice.ogg'
+	hitsound = SFX_WOUND_SPLATTER
```
Comment (aidan-bowman) [inline code/game/objects/items/weapons/material/material_weapons.dm]: Can't compare sounds right now, but are you sure this SFX is more appropriate than bladeslice.ogg? From the definition name, it sounds like it'd be better for a, well, splatter effect.

**Review by aidan-bowman — CHANGES_REQUESTED** [review]

File: code/game/objects/items/weapons/material/material_weapons.dm (hunk 4 above)
Comment (aidan-bowman) [reply code/game/objects/items/weapons/material/material_weapons.dm]: Listened to the sound and it seemed appropriate, ALTHOUGH you should really move this to the types that actually are knives. Currently this hitsound will be used for shit like ashtrays.

In fact, this should probably be set as an abstract type. Slap a `base_type = /obj/item/material` on there for me please 🥺

Comment (77seven77seven) [reply code/game/objects/items/weapons/material/material_weapons.dm]: > Listened to the sound and it seemed appropriate, ALTHOUGH you should really move this to the types that actually are knives. Currently this hitsound will be used for shit like ashtrays.
> 
> In fact, this should probably be set as an abstract type. Slap a `base_type = /obj/item/material` on there for me please 🥺

base_type is undefined

Comment (aidan-bowman) [reply code/game/objects/items/weapons/material/material_weapons.dm]: > > Listened to the sound and it seemed appropriate, ALTHOUGH you should really move this to the types that actually are knives. Currently this hitsound will be used for shit like ashtrays.
> > In fact, this should probably be set as an abstract type. Slap a `base_type = /obj/item/material` on there for me please 🥺
> 
> base_type is undefined

sorry, `abstract_type`

Comment (77seven77seven) [reply code/game/objects/items/weapons/material/material_weapons.dm]: ok done

**Review by aidan-bowman — APPROVED** [review]

**Comment by 77seven77seven** [comment]
pls merg

File: code/game/objects/items/weapons/material/material_weapons.dm [hunk 5]
```diff
@@ -2,13 +2,15 @@
...
 /obj/item/material
-	hitsound = 'sounds/weapons/bladeslice.ogg'
 	gender = NEUTER
 	throw_speed = 3
 	throw_range = 7
 	w_class = ITEM_SIZE_NORMAL
 	health_max = 10
 
+	abstract_type = /obj/item/material
+
```
Comment (TheDarkElites) [inline code/game/objects/items/weapons/material/material_weapons.dm:12]: Whats with the spaces?

**Review by TheDarkElites — CHANGES_REQUESTED** [review]

File: code/game/objects/items/weapons/material/material_weapons.dm (hunk 5 above)
Comment (aidan-bowman) [reply code/game/objects/items/weapons/material/material_weapons.dm:12]: also, abstract_type should be right below the type itself

Comment (TheDarkElites) [inline code/game/objects/items/weapons/material/material_weapons.dm:12]: Get rid of the extra spaces

File: code/game/objects/items/weapons/material/material_weapons.dm [hunk 6]
```diff
@@ -2,13 +2,15 @@
...
 // They are also fragile based on material data and many can break/smash apart.
 /obj/item/material
-	hitsound = 'sounds/weapons/bladeslice.ogg'
 	gender = NEUTER
 	throw_speed = 3
 	throw_range = 7
 	w_class = ITEM_SIZE_NORMAL
 	health_max = 10
 
+	abstract_type = /obj/item/material
```
Comment (TheDarkElites) [inline code/game/objects/items/weapons/material/material_weapons.dm:11]: Abstract type should be put right below the type itself

**Review by TheDarkElites — CHANGES_REQUESTED** [review]

File: code/game/objects/items/weapons/material/material_weapons.dm (hunk 6 above)
Comment (Tsurupeta) [reply code/game/objects/items/weapons/material/material_weapons.dm:11]: Unresolved

**Comment by Tsurupeta** [comment]
Resolve change requests and I'll merge it

**Comment by TheDarkElites** [comment]
John cant come to the phone right now

(If you want this merged just re-open with the review fixed)

**Pull request closed without merging**